Automattic/wp-calypso
# Checkoutv2: Term variant picker dropdown

This PR refactors the dropdown picker found in checkout so that it fits into the narrower sidebar format and handles the option text elegantly.

Related to https://github.com/Automattic/payments-shilling/issues/1969

## Proposed Changes

**Dotcom plan - closed picker**
| Before | After |
| ----- | ----- |
| ![image](https://github.com/Automattic/wp-calypso/assets/16580129/2e337404-978a-4cb4-b503-ac82757510e1) | ![image](https://github.com/Automattic/wp-calypso/assets/16580129/a7a60e43-9c99-43f6-8d64-181017bd6f45) |

**Dotcom plan - open picker**
| Before | After |
| ----- | ----- |
| ![image](https://github.com/Automattic/wp-calypso/assets/16580129/d9ad310e-70b2-4950-b39f-d19a3df8db9a) | ![image](https://github.com/Automattic/wp-calypso/assets/16580129/86266ab4-2c63-4041-b06f-b13c505bd9ab) |

**Jetpack product - open picker**
| Before | After |
| ----- | ----- |
| ![image](https://github.com/Automattic/wp-calypso/assets/16580129/98f1ba2a-2a7b-495b-a299-f829cbf0b764) | ![image](https://github.com/Automattic/wp-calypso/assets/16580129/b4d76db8-a4d3-40ca-9b6c-af339c22bc42) |

## Testing Instructions

* Go to Checkout without any query parameters and ensure the original, expected, behavior still shows (compare to production), you are looking for any styling leaks from the v2 design into v1
* Add the query param to the checkout URL `checkoutVersion=2` and test the new dropdown design
* Check for any unusual text flow or overwrap
* Check that the dropdown doesn't resize and stays around 200px wide
* Check to make sure no text/pricing/sale information is missing
* Change term lengths and make sure the cart updates the product to the correct billing term
